pkg/retry/retry.go

```diff
@@ -81,3 +83,31 @@ func WithBackoff(
 		}
 	}
 }
+
+// Opinionated considers timeouts, temporary, DNS and connection refused errors as retryable.
+func Opinionated(err error) bool {
```

Comment:
I don't like the name, doesn't really say what it does. Why not simply `IsRetryable`, maybe adding something like "common" or "generic" (to state that it catches the most common cases and others may choose to extend it)?